Implement buttons states

pr0t3us opened the pull request:
1. Added background selectors for the `Button` and `Button.Black` style.
2. Added `ViewsValidationButton` which should validate the fields based on the set validators (via `OnChangeValidator` interface). 
3. Created several validators to handle different view types: `RadioGroupsValidator`, `SpinnersValidator` and `TextViewsValidator`.
4. Used `RadioGroupsValidator` and `TextViewsValidator` to handle views in `BranchDetailsFragment`, `SpinnersValidator` and `TextViewsValidator` in `BranchSelectionFragment` and `TextViewsValidator` in `NoteFragment`

Fixes #47

pr0t3us committed:
Create dark background selector for button

pr0t3us committed:
Use ValidationInputButton widget to check against non empty fields

pr0t3us committed:
Create default button background selector

pr0t3us committed:
Create ViewsValidationButton together with OnChangeValidator interface

pr0t3us committed:
Replace ValidationInputButton with ViewsValidationButton

pr0t3us committed:
Use ViewsValidationButton to validate fields in BranchDetailsFragment

pr0t3us committed:
Use ViewsValidationButton to validate fields in NoteFragment

pr0t3us committed:
Validate views after setting validators

pr0t3us committed:
Use ViewsValidationButton to validate fields in BranchSelectionFragment

pr0t3us committed:
Remove validators from login button

AlexandraDaraban commented on app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginActivity.kt:
```diff
@@ -11,6 +11,7 @@ import ro.code4.monitorizarevot.helper.startActivityWithoutTrace
 import ro.code4.monitorizarevot.ui.base.BaseActivity
 import ro.code4.monitorizarevot.ui.branch.BranchActivity
 import ro.code4.monitorizarevot.ui.onboarding.OnboardingActivity
+import ro.code4.monitorizarevot.widget.validation.TextViewsValidator
```
Remove unused import

AlexandraDaraban commented on app/src/main/res/drawable/bg_button_dark.xml:
```diff
@@ -1,8 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<shape xmlns:android="http://schemas.android.com/apk/res/android"
-    android:padding="10dp"
-    android:shape="rectangle">
-    <!-- you can use any color you want I used here gray color-->
-    <solid android:color="?attr/colorAccent" />
-    <corners android:radius="4dp" />
-</shape>
```
Delete resource if not used

AlexandraDaraban commented on app/src/main/java/ro/code4/monitorizarevot/ui/branch/details/BranchDetailsFragment.kt:
```diff
@@ -81,12 +80,17 @@ class BranchDetailsFragment : BaseFragment<BranchViewModel>() {
+            environmentRadioGroup,
+            genderRadioGroup
+        )
+        val textViewsValidator = TextViewsValidator(arrivalTime, departureTime)
```
Departure time is optional and should be removed from here

AlexandraDaraban reviewed (changes requested):
Hey, @pr0t3us. Thank you for your contribution. The PR looks good, however there are some issues that I would like you to address before merging the PR. Also, I cloned your repo locally and I noticed the text of the buttons doesn't change when the button is disabled. Could you look more into this, please? Thanks 😄

pr0t3us committed:
Update alpha channel for the button color on disabled state

pr0t3us committed:
Remove unused drawable

pr0t3us committed:
Remove departure time from validation

pr0t3us commented:
Hi @AlexandraDaraban thank you for your fair review. I've addressed your comments. Looks like I've imported the wrong alpha channel value for the disabled button color, also I suggest using a more pronounced color for the normal state.

pr0t3us committed:
Use ViewsValidationButton for login layout after rebase

AlexandraDaraban commented:
Closed in #115
